internal/controller/component_customizer.go

```diff
@@ -73,25 +73,41 @@ func customizeObjectsFn(provider operatorv1.GenericProvider) func(objs []unstruc
 					}))
 			}
 
+			//nolint:nestif
```

Comment:
Not sure if a nolint is allowed here. I'll have to create a few helper functions if not. lmk which approach is better!